specutils/wcs/specwcs.py

```diff
@@ -395,12 +395,41 @@ class CompositeWCS(Model):
     wcs_list : list of callable objects, optional
         The object's wcs_list will be instantiated using wcs from this list
     """
-    def __init__(self, wcs_list=None):
+    _default_equivalencies = u.spectral()
+
+
+    def __init__(self, wcs_list=None, unit=None):
+        self._unit = None
+
         self.wcs_list = []
         if wcs_list is not None:
             for wcs in wcs_list:
                 self.add_WCS(wcs)
 
+    @property
+    def unit(self):
+        if self._unit is None:
+            return 1.0
+        else:
+            return self._unit
+
+    @unit.setter
+    def unit(self, value):
+        if value is None:
+            warnings.warn(
+                'Initializing a Spectrum1D WCS with units set to `None` is not recommended')
+            self._unit = None
+        else:
+            self._unit = u.Unit(value)
+
+    @property
+    def equivalencies(self):
```

Comment:
I found the test would fail without them.  I believe
it is because CompositeWCS is not a BaseSpectrum1DWCS.
I don’t think *any* CompositeWCS would work as coded (and
I couldn’t find any test that used it).
I suspect there is a better ‘work around’ but this was my best
while I tried to get the DC-FLAG aspect integrated.

I’ll make the changes for DC-FLAG you suggest. 

X


On May 16, 2015, at 7:34 AM, Wolfgang Kerzendorf <notifications@github.com> wrote:

> In specutils/wcs/specwcs.py:
> 
> > +        if self._unit is None:
> > +            return 1.0
> > +        else:
> > +            return self._unit
> > +
> > +    @unit.setter
> > +    def unit(self, value):
> > +        if value is None:
> > +            warnings.warn(
> > +                'Initializing a Spectrum1D WCS with units set to `None` is not recommended')
> > +            self._unit = None
> > +        else:
> > +            self._unit = u.Unit(value)
> > +
> > +    @property
> > +    def equivalencies(self):
> what are these changes for?
> 
> —
> Reply to this email directly or view it on GitHub.
> 
> !DSPAM:826,555755922149227501926!

----------------------------------------------
J. Xavier Prochaska
UCO/Lick Observatory
1156 High St.
UC Santa Cruz
Santa Cruz, CA 95064

http://www.ucolick.org/~xavier/
http://imps.ucolick.org/
http://www.qpqsurvey.org/
http://k1dm3.ucolick.org/
http://www.ucolick.org/~xavier/IDL/index.html
https://github.com/profxj/xastropy

831-459-2135 (Direct)
831-459-2991 (UCO/Lick Main)
831-459-5244 (Fax)